object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -608,4 +612,29 @@ private static String capitalize(String prop) {
    */
   private static final ImmutableSet<String> IGNORED_METHOD_NAMES =
       ImmutableSet.of("equals", "hashCode", "toString", "<init>", "toBuilder");
+
+  /** Taken from AutoValue source code */
+  private static final ImmutableSet<String> OPTIONAL_CLASS_NAMES =
+      ImmutableSet.of(
+          "com.".concat("google.common.base.Optional"), // subterfuge to foil shading
```

Comment:
Do we really want to keep this in our code? I'm sure that they had a good reason to do it, but it is very confusing here.